log/packetlogger.go

```diff
@@ -0,0 +1,155 @@
+package log
+
+import (
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithLogLevel sets the log level
+func WithLogLevel(level string) LoggerOption {
+	return func(args *PacketLogger) { args.LogLevel = level }
+}
+
+// WithOutputPaths adds output paths
+func WithOutputPaths(paths []string) LoggerOption {
+	return func(args *PacketLogger) { args.OutputPaths = paths }
+}
+
+// WithServiceName adds a service name a logged field
+func WithServiceName(name string) LoggerOption {
+	return func(args *PacketLogger) { args.ServiceName = name }
+}
+
+// WithKeysAndValues adds extra key/value fields
```

Comment:
adds seems like the wrong word, I read that as "append" but this is overwriting. Can we get an append instead of overwrite instead?

Reply:
sorry, not following. `WithKeysAndValues`  is used to set `zapConfig.InitialFields`. I'm not seeing where the overwrite is happening, sorry.

Reply:
There's no guard against someone doing:
```go
NewPacketLogger(WithKeysAndValues(map[string]interface{"foo": "bar"}), WithKeysAndValues(map[string]interface{"bar": "baz"}))
```

Reply:
copy that, will make this additive.